sparkplug-core/src/clojure/sparkplug/rdd.clj

```diff
@@ -55,23 +56,16 @@
   (first (remove internal-call? (.getStackTrace (Exception.)))))
 
 
-(defn- unmangle
-  "Given the name of a class that implements a Clojure function, returns the
-  function's name in Clojure.
-
-  If the true Clojure function name contains any underscores (a rare
-  occurrence), the unmangled name will contain hyphens at those locations
-  instead."
-  [classname]
-  (-> classname
-      (str/replace #"^(.+)\$(.+)(|__\d+)$" "$1/$2")
-      (str/replace \_ \-)))
+(defn- demunge
+  "Demunge a class name emitted by the Clojure compiler."
+  [^String classname]
+  (Compiler/demunge classname))
```

Comment:
Seems simpler to call `Compiler/demunge` directly in the two callsites below and omit this function now.